app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling like requests.
+    class LikesController < ApplicationController
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!
+      before_action :like, only: %i[destroy]
+      before_action :authorize_like!
+      after_action :verify_authorized
+
+      def index
+        if params[:post].blank?
```

Comment:
1) post_id instead of post
2) param presence doesn't mean resource presence. we need to make sure resource exists in our db first, and then pull likes.

```
post = Post.find_by(id: ...)

if post
  render json: post.likes
else
  ... 404
end